prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -1266,7 +1273,10 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
     }
 
     /// Encode the lhs and the rhs of the assignment that create the loan
-    fn encode_loan_places(&self, loan_places: &LoanPlaces<'tcx>) -> (vir::Expr, vir::Expr, bool) {
+    fn encode_loan_places(
+        &self,
+        loan_places: &LoanPlaces<'tcx>
+    ) -> (PlaceEncoding, PlaceEncoding, bool) {
```

Comment:
Is every usage of `encode_loan_places` going to call `.expect_expr()` on the returned `PlaceEncoding`? If so, what about calling `.expect_expr()` inside this function, returning `vir::Expr`?

Reply:
Even better, we could add a new `ProcedureEncoder::encode_place` with the same signature of the *old* `MirEncoder::encode_place`, replacing every `self.mir_encoder.encode_place` call with `self.encode_place`. The conversion from `PlaceEncoding` to `vir::Expr` could then be implemented just in `ProcedureEncoder::encode_place`.

Reply:
what `vir::Expr` would we return for an array access then?

Reply:
Adding `Vec<vir::Stmt>` to the return type of `ProcedureEncoder::encode_place` is needed for that.